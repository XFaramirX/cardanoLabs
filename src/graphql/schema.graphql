# import Image,Artist from "schema/homepage.graphql"

type Query {
  getAllArtist: [Artist]
  getArtist(id: ID!): Artist
  getImages: [Image]
}
