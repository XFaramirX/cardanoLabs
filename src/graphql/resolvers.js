const { ApolloError } = require("apollo-server");

// Firebase App (the core Firebase SDK) is always required and must be listed before other Firebase SDKs
const admin = require("firebase-admin");
const serviceAccount = require("../../service-account-file.json");

admin.initializeApp({
  credential: admin.credential.cert(serviceAccount),
  databaseURL: "https://angelbernalesquivel-ecf4b-default-rtdb.firebaseio.com",
});
// Resolver map

const db = admin.firestore();

async function getImages(parent, args, context, info) {
  const imagesRef = await db.collection("images").get();
  try {
    if (!imagesRef.docs[0].exists) {
      // No document exist
    } else {
      return imagesRef.docs.map((image) => image.data());
    }
  } catch (error) {
    throw new ApolloError(error);
  }
}

async function getArtist(parent, args, context, info) {
  const artists = await admin
    .firestore()
    .collection("artist")
    .doc(args.id)
    .get();

  try {
    if (!artists.exists) {
      // No document exist
    } else {
      return artists.data();
    }
  } catch (error) {
    throw new ApolloError(error);
  }
}

async function getAllArtist(parent, args, context, info) {
  try {
    const artistList = await admin.firestore().collection("artist").get();
    return artistList.docs.map((artist) => artist.data());
  } catch (error) {
    throw new ApolloError(error);
  }
}
async function getImagesbyArtist(parent, args, context, info) {
  const imagesRef = await db.collection("images").get();
  try {
    const result = imagesRef.docs.map((image) => image.data());
    return result.filter((res) => res.createdBy == parent.name);
  } catch (error) {
    throw new ApolloError(error);
  }
}

const resolvers = {
  Query: {
    getAllArtist,
    getArtist,
    getImages,
  },
  Artist: {
    images: getImagesbyArtist,
  },
};

export default resolvers;
