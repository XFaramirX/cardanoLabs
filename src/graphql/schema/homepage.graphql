type Artist {
  id: ID!
  name: String
  instagram: String
  images: [Image]
}

type Image {
  id: ID
  title: String
  url: String!
  attributes: [String]
  description: String
  createdBy: String
}
