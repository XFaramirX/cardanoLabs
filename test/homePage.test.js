import "core-js/stable";
import "regenerator-runtime/runtime";

import { join } from "path";
import { loadSchemaSync } from "@graphql-tools/load";
import { GraphQLFileLoader } from "@graphql-tools/graphql-file-loader";
import { addResolversToSchema } from "@graphql-tools/schema";
const { ApolloServer, gql, MockList } = require("apollo-server");
const { createTestClient } = require("apollo-server-testing");

const schema = loadSchemaSync(
  join(__dirname, "../src/graphql/schema.graphql"),
  {
    loaders: [new GraphQLFileLoader()],
  }
);

import resolvers from "../src/graphql/resolvers";

const schemaWithResolvers = addResolversToSchema({
  schema,
  resolvers,
});

const server = new ApolloServer({
  schema: schemaWithResolvers,
  graphiql: true,
  // mocks,
  // mockEntireSchema: false,
});

const { query, mutate } = createTestClient(server);

const mocks = {
  Query: () => ({
    images: () => new MockList([6, 9]),
  }),
  Image: () => ({
    title: () => "Astro Kitty, Space Explorer 01",
    url: () =>
      "https://res.cloudinary.com/dety84pbu/image/upload/v1606816219/kitty-veyron-sm_mctf3c.jpg",
  }),
};

it("Fetches All Images", async () => {
  const GET_IMAGES = gql`
    query {
      getImages {
        id
        title
        url
        attributes
        description
        createdBy
      }
    }
  `;

  const response = await query({ query: GET_IMAGES });
  expect(response).toMatchSnapshot();
});

it("Fetches All Artist", async () => {
  const GET_ALL_ARTIST = gql`
    query {
      getAllArtist {
        id
        name
        instagram
        images {
          id
          title
          url
          attributes
          description
          createdBy
        }
      }
    }
  `;

  const response = await query({ query: GET_ALL_ARTIST });
  expect(response).toMatchSnapshot();
});

it("Fetches Artist By ID", async () => {
  const GET_ALL_ARTIST = gql`
    query {
      getArtist(id: "UVrxdwKDO4MmjXizLSjV") {
        id
        name
        instagram
        images {
          id
          title
          url
          attributes
          description
          createdBy
        }
      }
    }
  `;

  const response = await query({ query: GET_ALL_ARTIST });
  expect(response).toMatchSnapshot();
});
