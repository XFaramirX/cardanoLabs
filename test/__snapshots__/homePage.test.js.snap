// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fetches All Artist 1`] = `
Object {
  "data": Object {
    "getAllArtist": Array [
      Object {
        "id": "1",
        "images": Array [
          Object {
            "attributes": Array [
              "Oleo Painting",
              "Rarity 5 out of 7",
              "Edition 1 out of 12",
              "Colombia",
            ],
            "createdBy": "BernalEsquivel01",
            "description": "Oleo Painting made by artist angel bernal colombia bogota. Selling as NFT",
            "id": "01",
            "title": "BernalEsquivel01",
            "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/BestLandscape.jpg?alt=media&token=e25df19b-4818-4adb-ab70-8143949a0d87",
          },
          Object {
            "attributes": null,
            "createdBy": "BernalEsquivel01",
            "description": null,
            "id": "04",
            "title": "Bernal04",
            "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/OleoPainting.jpg?alt=media&token=44836ff5-38a8-45ba-9253-c5cbb51427f3",
          },
        ],
        "instagram": "https://www.facebook.com/David.B.Bernal/",
        "name": "BernalEsquivel01",
      },
      Object {
        "id": "2",
        "images": Array [
          Object {
            "attributes": null,
            "createdBy": "EvaAcosta",
            "description": "    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi delectus magnam temporibus, adipisci cupiditate libero soluta, sint sed velit ullam minima aperiam veniam? Maxime id eligendi quia, tempora quis facilis!",
            "id": "03",
            "title": "Bernal03",
            "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/MountainWarrior2005-2017.jpg?alt=media&token=9b398009-df06-4c9f-8f6c-190c735fbf8e",
          },
        ],
        "instagram": null,
        "name": "EvaAcosta",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Fetches All Images 1`] = `
Object {
  "data": Object {
    "getImages": Array [
      Object {
        "attributes": Array [
          "Oleo Painting",
          "Rarity 5 out of 7",
          "Edition 1 out of 12",
          "Colombia",
        ],
        "createdBy": "BernalEsquivel01",
        "description": "Oleo Painting made by artist angel bernal colombia bogota. Selling as NFT",
        "id": "01",
        "title": "BernalEsquivel01",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/BestLandscape.jpg?alt=media&token=e25df19b-4818-4adb-ab70-8143949a0d87",
      },
      Object {
        "attributes": null,
        "createdBy": "EvaAcosta",
        "description": "    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi delectus magnam temporibus, adipisci cupiditate libero soluta, sint sed velit ullam minima aperiam veniam? Maxime id eligendi quia, tempora quis facilis!",
        "id": "03",
        "title": "Bernal03",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/MountainWarrior2005-2017.jpg?alt=media&token=9b398009-df06-4c9f-8f6c-190c735fbf8e",
      },
      Object {
        "attributes": null,
        "createdBy": null,
        "description": null,
        "id": "05",
        "title": "Bernal05",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/Symbol.jpg?alt=media&token=600e4c09-55ca-4b0b-bcac-88e0308c6f81",
      },
      Object {
        "attributes": null,
        "createdBy": null,
        "description": null,
        "id": "02",
        "title": "bernal02",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/CaminantesPutimaes.jpg?alt=media&token=4e416115-b3c0-4cbe-b0bc-c10b2259e193",
      },
      Object {
        "attributes": null,
        "createdBy": "BernalEsquivel01",
        "description": null,
        "id": "04",
        "title": "Bernal04",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/OleoPainting.jpg?alt=media&token=44836ff5-38a8-45ba-9253-c5cbb51427f3",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Fetches Artist By ID 1`] = `
Object {
  "data": Object {
    "getArtist": Object {
      "id": "2",
      "images": Array [
        Object {
          "attributes": null,
          "createdBy": "EvaAcosta",
          "description": "    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi delectus magnam temporibus, adipisci cupiditate libero soluta, sint sed velit ullam minima aperiam veniam? Maxime id eligendi quia, tempora quis facilis!",
          "id": "03",
          "title": "Bernal03",
          "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/MountainWarrior2005-2017.jpg?alt=media&token=9b398009-df06-4c9f-8f6c-190c735fbf8e",
        },
      ],
      "instagram": null,
      "name": "EvaAcosta",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Fetches Images 1`] = `
Object {
  "data": Object {
    "getImages": Array [
      Object {
        "attributes": Array [
          "Oleo Painting",
          "Rarity 5 out of 7",
          "Edition 1 out of 12",
          "Colombia",
        ],
        "createdBy": "BernalEsquivel01",
        "description": "Oleo Painting made by artist angel bernal colombia bogota. Selling as NFT",
        "id": "01",
        "title": "BernalEsquivel01",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/BestLandscape.jpg?alt=media&token=e25df19b-4818-4adb-ab70-8143949a0d87",
      },
      Object {
        "attributes": null,
        "createdBy": "EvaAcosta",
        "description": "    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nisi delectus magnam temporibus, adipisci cupiditate libero soluta, sint sed velit ullam minima aperiam veniam? Maxime id eligendi quia, tempora quis facilis!",
        "id": "03",
        "title": "Bernal03",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/MountainWarrior2005-2017.jpg?alt=media&token=9b398009-df06-4c9f-8f6c-190c735fbf8e",
      },
      Object {
        "attributes": null,
        "createdBy": null,
        "description": null,
        "id": "05",
        "title": "Bernal05",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/Symbol.jpg?alt=media&token=600e4c09-55ca-4b0b-bcac-88e0308c6f81",
      },
      Object {
        "attributes": null,
        "createdBy": null,
        "description": null,
        "id": "02",
        "title": "bernal02",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/CaminantesPutimaes.jpg?alt=media&token=4e416115-b3c0-4cbe-b0bc-c10b2259e193",
      },
      Object {
        "attributes": null,
        "createdBy": "BernalEsquivel01",
        "description": null,
        "id": "04",
        "title": "Bernal04",
        "url": "https://firebasestorage.googleapis.com/v0/b/angelbernalesquivel-0001.appspot.com/o/OleoPainting.jpg?alt=media&token=44836ff5-38a8-45ba-9253-c5cbb51427f3",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
